wasm/src/main/scala/ir2wasm/WasmExpressionBuilder.scala

```diff
@@ -631,6 +732,13 @@ private class WasmExpressionBuilder private (
           case Some(primReceiverType) =>
             if (t.receiver.tpe == primReceiverType) {
               genTreeAuto(t.receiver)
+            } else if (t.receiver.isInstanceOf[IRTrees.This]) {
+              // TODO: investigate what's going on
+              // Don't know why, but it seems that `this` isn't boxed even if
+              // t.receiver.tpe = ClassType(ClassName<java.lang.Boolean>), and
+              // primReceiverType = BooleanType
+              // This wired patch is required for `(func $f#java.lang.Boolean#compareTo_Ljava.lang.Boolean_R`
```

Comment:
Without this change, validation fails because of the following wasm code

```wasm
  (func $f#java.lang.Boolean#compareTo_Ljava.lang.Boolean_R (type $f.268)
     (param $___<this> i32) (param $that anyref) (result anyref)
    
     local.get $___<this>
     ;; <this> is already unboxed
     ref.as_non_null
     call $__scalaJSHelpers#uZ
     local.get $that
     call $f#java.lang.Boolean#compareTo_Ljava.lang.Boolean_I
     call $__scalaJSHelpers#bI)
```

🤔